src/App.tsx

```diff
@@ -1,27 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import 'bulma/css/bulma.css';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  // const [currentItems, setCurrentItems] = useState(items);
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const lastItemIndex = perPage * currentPage;
+  const firstItemIndex = lastItemIndex - perPage;
+  const preparedItems = items.slice(firstItemIndex, lastItemIndex);
+
+  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setCurrentPage(1);
+    setPerPage(+e.target.value);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {firstItemIndex + 1} -{' '}
```

Comment:
The current implementation will display 'items 1 - 0' when there are no items. You should handle this edge case to display a more user-friendly message.